docs/design-proposals/cloud-edge-communication/cloud-edge-communication-framework.md

```diff
@@ -11,18 +11,255 @@ Because the cloud and edge is located in the network boundary of different secur
+		"virtualPresenceIPcidr": "192.168.0.0/24",
+		"eSiteName": "TokyoChibaSite1"
+	}
+	
+	{
+	    "kind": "VirtualPresence",
+	    "apiVersion": "v1",
+	    "metadata": {
+	        "name": "vp1"
+		},
+		"virtualIp": "192.168.0.10",
+		"eServiceName": "TykyoChibaNginx"
+	}
+	
+	{
+	    "kind": "EService",
+	    "apiVersion": "v1",
+	    "metadata": {
+	        "name": "TykyoChibaNginx"
+		},
+		"port": 8080,
+		"protocol": "TCP",
+		"ip": "10.0.0.10",
+		"domainName": "EServiceNginx"
+		"eSiteName": "TokyoChibaSite1"
+	}
+	
+	{
+	    "kind": "EServer",
+		"apiVersion": "v1",
+		"eServerName": "Server1",
+		"ip": "10.0.0.20",
+		"eSiteName": "TokyoChibaSite1"
+	}
+	
+	{
+	    "kind": "EPolicy",
+	    "apiVersion": "v1",
+	    "metadata": {
+	        "name": "Policy1"
+		},
+		"allowedServers": ["Server1"]
+	}
+	
+	{
+	    "kind": "ServiceExpose",
+	    "apiVersion": "v1",
+	    "eServiceName": "TykyoChibaNginx",
+		"dnsName": "url1.test.com",
+	    "eSites": ["TokyoChibaSite1"],
+	    "ePolicys": ["Policy1", "Policy2"]
+	}
+
+
+#### API Structs
+
+Here are the ESite data struct introduced into arktos，Represent each site, and the central cloud region also belongs to the site：
+
+	type ESite struct {
+		metav1.TypeMeta
+		metav1.ObjectMeta
+	}
+
+EGateway data struct：
+
+	type EGateway struct {
+		metav1.TypeMeta
+		metav1.ObjectMeta
+	
+		// Ip address of the gateway
+		Ip string
+	
+		// Virtual presence ip address cidr of this gateway
+		VirtualPresenceIPcidr string
+	
+		// ESiteName associated to the gateway
+		ESiteName string
+	}
+
+VirtualPresence data struct：
+
+	type VirtualPresence struct {
+		metav1.TypeMeta
+		metav1.ObjectMeta
+	
+		// Ip address of the virtual presence
+		VirtualIp string
+	
+		// Associated service name
+		EServiceName string
+	}
+
+EService data struct：
+
+	type EService struct {
+		metav1.TypeMeta
+		metav1.ObjectMeta
+	
+		// Protocol of the service
+		Protocol string
+	
+		// Port of the service use
+		Port int32
+	
+		// Ip of the service
+		Ip string
+
+		// DomainName of service
+		DomainName string
+	
+		// Associated site name
+		ESiteName string
+	}
+
+EServer data struct：
+
+	type EServer struct {
+		metav1.TypeMeta
+		metav1.ObjectMeta
+	
+		// EServerName of the server
+		EServerName string
+	
+		// Ip of the service
+		Ip string
+	
+		// Associated site name
+		ESiteName string
+	}
+
+EPolicy data struct：
+
+	type EPolicy struct {
+		metav1.TypeMeta
+		metav1.ObjectMeta
+	
+		// Allowed server names of this policy
+		AllowedServers []string
+
+		// Allowed site names of this policy
+		AllowedSites []string
+	}
+
+ServiceExpose data struct：
+
+	type ServiceExpose struct {
+		metav1.TypeMeta
+		metav1.ObjectMeta
+	
+		// Associated service name will be exposed
+		EServiceName string
+	
+		// Dns Name of the service will be exposed
+		DnsName string
+	
+		// ESite name list
+		ESites []string
+	
+		// EPolicys name list of the service will be allowed
+		EPolicys []string
+	}
+
+## Implementation Steps：
```

Comment:
so at this point which proposal is being used?

Reply:
Proposal B, I will add it before the implementation steps, thx